ros_buildfarm/status_page.py

```diff
@@ -530,6 +530,296 @@ def _get_comparable_loose_versions(version_str1, version_str2):
+
+        if len(repos_blocked_by) > 0:
+            tags += _filter_tag_wrap('BLOCKED')
+        elif len(repos_blocking) > 0:
+            tags += _filter_tag_wrap('UNBLOCKED_BLOCKING')
+        else:
+            tags += _filter_tag_wrap('UNBLOCKED_UNBLOCKING')
+
+    url = data.get('url', None)
+    if url:
+        name_html = '<a href={0}>{1}</a>'.format(url, name)
+    else:
+        name_html = name
+    name_cell = name_html + tags
+
+    # repo name cell
+    row['name'] = _table_cell_wrap(name_cell, name)
+
+    # whether or not the repo has been released
+    row['version'] = _table_cell_wrap(data.get('version', 'Not released'))
+
+    # num repos blocked by
+    row['num_repos_blocked_by'] = _table_cell_wrap(len(repos_blocked_by))
+
+    # repos blocked by
+    row['repos_blocked_by'] = _table_cell_wrap('<br>'.join(
+        _name_query_wrap(repo) for repo in sorted(repos_blocked_by.keys())))
+
+    # maintainers info
+    maintainers = data.get('maintainers', {})
+    maintainers_cell = ''
+    for repo_name in sorted(maintainers.keys()):
+        maintainers_cell += repo_name + ':<br>'
+        maintainers_cell += '<br>'.join(
+                '&emsp;<a href="mailto:{0}">{1}</a>'
+                .format(maintainers[repo_name][name], name)
+                for name in sorted(maintainers[repo_name].keys()))
+        maintainers_cell += '<br>'
+    row['maintainers_of_repos_blocked_by'] = _table_cell_wrap(maintainers_cell)
+
+    # num repos blocking recursively
+    row['num_repos_recursively_blocked'] = _table_cell_wrap(
+            len(data.get('recursive_repos_blocking', [])))
+
+    # num repos blocking
+    row['num_repos_blocked'] = _table_cell_wrap(len(repos_blocking))
+
+    # repos blocking by not being released
+    row['repos_blocked'] = _table_cell_wrap('<br>'.join(
+        _name_query_wrap(repo) for repo in sorted(repos_blocking)))
+
+    return row
+
+
+def _is_released(repo, dist_file):
+    return repo in dist_file.repositories and \
+        dist_file.repositories[repo].release_repository is not None and \
+        dist_file.repositories[repo].release_repository.version is not None
+
+
+def _get_blocked_releases_info(
+        rosdistro_name, repo_names=None, depth=1):
+    import rosdistro
+    from rosdistro.dependency_walker import DependencyWalker
+    from catkin_pkg.package import InvalidPackage, parse_package_string
+
+    prev_rosdistro_name = None
+
+    index = rosdistro.get_index(rosdistro.get_index_url())
+    valid_rosdistro_names = list(index.distributions.keys())
+    valid_rosdistro_names.sort()
+    if rosdistro_name is None:
+        rosdistro_name = valid_rosdistro_names[-1]
+    print('Checking packages for "%s" distribution' % rosdistro_name)
+
+    # Find the previous distribution to the current one
+    try:
+        i = valid_rosdistro_names.index(rosdistro_name)
+    except ValueError:
+        print('Distribution key not found in list of valid distributions.')
```

Comment:
Error messages should go to `stderr`. Use `from __future__ import print_function` at the top of the file and then `print(..., file=sys.stderr)`.